src/Searchbar.vue

```diff
@@ -15,35 +15,47 @@ export default {
   },
   computed: {
     primitiveData() {
+      function getTotalMatches(searchTarget, regexes) {
+        return regexes.reduce((acc, regex) => (regex.test(searchTarget) ? acc + 1 : acc), 0);
+      }
+
       if (this.value.length < 2) {
         return [];
       }
       if (!this.data) {
         return undefined;
       }
       const matches = [];
-      const regexes = this.value.split(' ').map(searchKeyword => new RegExp(searchKeyword, 'i'));
+      const regexes = this.value.split(' ').filter(searchKeyword => searchKeyword !== '')
+        .map(searchKeyword => new RegExp(searchKeyword, 'i'));
       this.data.forEach((entry) => {
         const { headings, src, title } = entry;
         const keywords = entry.keywords || '';
-        let searchTarget = [title].concat(keywords).join(' ');
-        const isMatchingPage = regexes.every(regex => regex.test(searchTarget));
-        if (isMatchingPage) {
-          matches.push(entry);
-        }
-        Object.entries(headings).forEach(([id, text]) => {
-          searchTarget = [title].concat(keywords).concat(text).join(' ');
-          if (isMatchingPage || regexes.every(regex => regex.test(searchTarget))) {
-            matches.push({
-              heading: { id, text },
-              keywords,
-              src,
-              title,
-            });
+        let searchTarget = [title].concat(keywords).concat(Object.values(headings)).join(' ');
```

Comment:
I don't get the point of concatenating all headings.

Reply:
if not how do you wanna check isMatchingPage,

Reply:
But isMatchingPage is supposed to not check for headings right? The calculation of the ranking should be independent of isMatchingPage.

Reply:
`const isMatchingPage = getTotalMatches(searchTarget, regexes) === matchNum; `
It is comparing the totalMatches of all info(including all headings) with info (not including headings)

Reply:
Ok, how about this:

```javascript
    this.data.forEach((entry) => {
        const {headings, src, title} = entry;
        const keywords = entry.keywords || '';
        let searchTarget = [title].concat(keywords).join(' ');
        const pageTotalMatches = getTotalMatches(searchTarget, regexes);
        if (pageTotalMatches > 0) {
          matches.push(Object.assign(entry, { totalMatches: pageTotalMatches }));
        }
        Object.entries(headings).forEach(([id, text]) => {
          const searchTarget = [title].concat(keywords).concat(text).join(' ');
          const totalMatches = getTotalMatches(searchTarget, regexes);
          if (totalMatches > 0) {
            matches.push({
              heading: {id, text},
              keywords,
              src,
              title,
              totalMatches,
            });
          }
        });
      });
```

Reply:
but this will rank the page entry lower than its heading entry.

Reply:
That is fine, if it matches less keywords it should be lower anyway.

Reply:
If it matches only `title` and `keywords`, it will show an entry for every `heading`?
That does not seem right — and makes the results unnecessarily long.

I'm not sure about ranking the page entry lower than its heading entries.
The page should be the main target; heading entries serve as quick links.
It seems pointless to show the page entry at the bottom.